lstchain/io/event_selection.py

```diff
@@ -141,6 +145,35 @@ def apply_global_gh_cut(self, data):
         """
         return data[data["gh_score"] > self.global_gh_cut]
 
+    def update_fill_cuts(self, cut_table):
+        """
+        For an energy-dependent cut table, update the cuts for bins with number
+        of events fewer than the minimum number, for which pyirf uses a
+        constant fill_value, usually at the energy threshold limits, with cut
+        evaluated at the nearest bin with number of events more than the given
+        minimum.
+        """
+        cut_table_new = cut_table.copy()
+
+        low_event_bins = np.argwhere(
+            cut_table["n_events"] < self.min_event_p_en_bin
+        ).ravel()
+        high_event_bins = np.where(
+            cut_table["n_events"] >= self.min_event_p_en_bin
+        )[0]
```

Comment:
Why use np.argwhere first, then np.where? If I am not mistaken, the way you use them, the effect is the same.
The second one could be 
np.argwhere(cut_table["n_events"] >= self.min_event_p_en_bin).ravel()

Reply:
Also, using `np.where` just to get the indices is deprecated I think (at least it's slower than just using `np.nonzero`)

Reply:
> np.argwhere(a) is almost the same as np.transpose(np.nonzero(a)), but produces a result of the correct shape for a 0D array.

> The output of argwhere is not suitable for indexing arrays. For this purpose use nonzero(a) instead.

Reply:
Thanks for the suggestions! Changed to using `np.nonzero`